tool/tsh/kube.go

```diff
@@ -581,6 +581,49 @@ func newKubeCredentialsCommand(parent *kingpin.CmdClause) *kubeCredentialsComman
 	return c
 }
 
+func getKubeCredLockfilePath(homePath, proxy string) (string, error) {
+	profilePath := profile.FullProfilePath(homePath)
+	// tsh stores the profiles using the proxy host as the profile name.
+	profileName, err := utils.Host(proxy)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	return keypaths.KubeCredLockfilePath(profilePath, profileName), nil
+}
+
+var ErrKubeCredLockfileFound = trace.AlreadyExists("Having problems with relogin, please use 'tsh login/tsh kube login' manually")
+
+func takeKubeCredLock(ctx context.Context, homePath, proxy string) (func(bool), error) {
+	kubeCredLockfilePath, err := getKubeCredLockfilePath(homePath, proxy)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// If kube credentials lockfile already exists, it means last time kube credentials was called
+	// we had an error while trying to issue certificate, return an error asking user to login manually.
+	if _, err := os.Stat(kubeCredLockfilePath); err == nil {
+		log.Debugf("Kube credentials lockfile was found at %q, aborting.", kubeCredLockfilePath)
+		return nil, ErrKubeCredLockfileFound
+	}
```

Comment:
I don't think not expiring lock is a problem, that's what we want, to fight one specific problematic UX when users get a lot of tabs open from many separate login attempts, so we want to persist lockfile between tsh runs if we know we need it. But good point about temporary connection problem. I addressed it, moved creating lockfile to a bit later stage and we'll remove it in case of networking problem, so it now should be able to recover from temporary network error on its own.

Reply:
Network problems do tend to be transient, but this is just a heuristic. We have no good way to tell whether the error is transient. In general, we should assume all errors are transient, and therefore we should try again after some time has passed. I think we can implement expiring locks quite easily using file system timestamps.

The problem with asking the user to login manually is that they will very likely *NOT* see that message. In this scenario `tsh` is invoked in the background.

Reply:
Yep, I get your point and agree with general idea, but we need to take into account all circumstances. In this case retrying after a few minutes would just lead to having same problem we're trying to solve in the beginning - users will get lots of tabs with SSO retries after leaving machines unattended. I think being resilient to possible intermittent network problems when we can't even connect to Teleport proxy is a good balance, we don't need to overcomplicate the solution. 
And this is only relevant to situation when user's session expired and we're trying to renew it in an automated fashion. In most cases users have ability to see that message in the logs (Lens, which spawned the original issue; `watch kubectl ...`; vscode extension etc), but even without it, trying to login manually is one of the first natural troubleshooting steps, we're just trying to make it a bit more user friendly. And manual login isn't blocked by this lockfile.

Reply:
All right, we can always revisit this point in the future if needed.